forem/forem
# Update feed filter language & component (RFC#236)

## What type of PR is this? (check all applicable)


## Description

This PR addresses RFC: https://github.com/forem/rfcs/pull/236 (Update feed filter language & component).

The main strands are:

### 1. Feed filter option terminology has changed

The functionality in terms of how the filters work hasn't changed at all, but the words we apply to each have:

Feed -> Trending
Week -> This Week
Month -> This Month
Year -> This Year
Infinity -> All Time

The heading of the page now also changes when you select a filter e.g. instead of "Posts" you should see "Trending" by default.

### 2. Dropdown for feed filtering on larger screens

Where we previously showed the "navigation tabs" component, we now show a dropdown of the feed filter options. I've kept the breakpoints for this in line with the current implementation, with the result being that this dropdown is shown on all screens apart from portrait orientation mobile phones.

https://user-images.githubusercontent.com/20773163/129867217-385cb61e-f9da-416e-a67e-10a66290ee65.mp4

### 3. Modal drawer for feed filtering on smallest screens

As above, this should only be visible on portrait orientation mobile phone screen size. In this viewport, when you click the feed filter options button you should see a modal drawer slide in from the bottom with the navigation options.

https://user-images.githubusercontent.com/20773163/129867235-76bc5737-b2f3-4c3a-9e76-fdedd796e38d.mp4

### 4. Change to time filter / route names

The URLs and filters used to decide which articles to show has changed to match the new terminology

`/` -> No change
`/latest` -> `/top/most-recent`
`/week` -> `/top/this-week`
`/month` -> `top/this-month`
`/year` -> `/top/this-year`
`/infinity` -> `/top/all-time`

Redirects have been created to redirect the old routes to the new ones.

## Related Tickets & Documents

https://github.com/forem/rfcs/pull/236

## QA Instructions, Screenshots, Recordings

When testing this PR, we need to bear in mind:
- This applies to the home feed, but also the tag index (e.g. `/t/beginners`)
- The logged in and logged out feed are rendered differently, so we need to test both 🙂 
- We should test on real mobile devices to make sure we capture any UX nuance of the new drawer component

**We'll need to test:**
- Navigating between different feed filter options
- Navigating backwards and forwards from articles/other pages, to make sure the dropdowns/modal drawers continue to work as expected
- Viewing each version of the feed (filtered/unfiltered, logged in/logged out, home feed/tag index) to make sure we're getting the same results as before
- Checking that articles being paged in have the correct search params for each filter. We can check this in the devtools network tab, inspecting the request's "sort_by" and "published_at" parameters
- Visting the old URL routes and checking the browser shows them being redirected to the new ones


### UI accessibility concerns?

Most of the accessibility concerns have been addressed and tested as part of the previous PR introducing the new MobileDrawer component, but things to keep an eye on are:

- If you see the appropriate checkmark next to your currently selected navigation link, this means `aria-current="page"` is successfully set. If you don't see a checkmark next to the link for your current page, something's gone wrong!
- The dropdown on larger screens should conform to the [usual dropdown pattern documented in storybook](https://storybook.forem.com/?path=/story/components-dropdowns--default)
- The mobile navigation drawer uses the new `MobileDrawer` under the hood, and expected interaction pattern in terms of focus management is the same as with a "normal" Modal. [Details again in storybook](https://storybook.forem.com/?path=/story/components-mobiledrawer--default)

## Added/updated tests?

      have not been included_

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Admin Guide](https://admin.forem.com/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![alt_text](gif_link)
